topologytest/hlv_test.go

```diff
@@ -72,9 +135,47 @@ func TestHLVCreateDocumentSingleActor(t *testing.T) {
 		t.Run(tc.description(), func(t *testing.T) {
 			peers, _ := setupTests(t, tc.topology, tc.activePeerID)
 
+			docID := tc.docID("")
 			docBody := []byte(fmt.Sprintf(`{"peer": "%s", "topology": "%s"}`, tc.activePeerID, tc.description()))
-			docVersion := peers[tc.activePeerID].CreateDocument(tc.collectionName(), tc.docID(), docBody)
-			waitForVersionAndBody(t, tc, peers, docVersion, docBody)
+			docVersion := peers[tc.activePeerID].CreateDocument(tc.collectionName(), tc.docID(""), docBody)
+			bodyVersion := BodyAndVersion{
+				docMeta:    docVersion,
+				body:       docBody,
+				updatePeer: tc.activePeerID,
+			}
+			waitForVersionAndBody(t, tc, peers, docID, bodyVersion)
+		})
+	}
+}
+
+// TestHLVCreateDocumentMultiActorConflict:
+//   - Create conflicting docs on each peer
+//   - Wait for docs last write to be replicated to all other peers
+func TestHLVCreateDocumentMultiActorConflict(t *testing.T) {
+	base.SetUpTestLogging(t, base.LevelDebug, base.KeyCRUD, base.KeyImport, base.KeyVV)
+	if base.UnitTestUrlIsWalrus() {
+		t.Skip("Panics against rosmar, CBG-4378")
+	} else {
+		t.Skip("Flakey failures on multi actor conflicting writes, CBG-4379")
+	}
+	for _, tc := range getMultiActorTestCases() {
+		if strings.Contains(tc.description(), "CBL") {
+			// Test case flakes given the WaitForDocVersion function only waits for a docID on the cbl peer. We need to be
+			// able to wait for a specific version to arrive over pull replication
+			t.Skip("We need to be able to wait for a specific version to arrive over pull replication, CBG-4257")
+		}
+		t.Run(tc.description(), func(t *testing.T) {
+			peers, replications := setupTests(t, tc.topology, "")
+
+			stopPeerReplications(t, replications)
```

Comment:
Is there a reason to stop replications before creating conflicting docs? Is this because `CreateDocument` can fail?

I think this is a good idea to debug tests at first but I think we'll want these writes to occur eventually while peers are active. I think this is worth marking as such and filing a followup ticket.

I think it might improve readability if we provide something like options to `setupTests` to say `startReplications` and `activePeer`. I am a big fan of `setupTestsOptions` structs so it's easy to understand what arguments like bools and strings mean.

 Actually I think `activePeer` might be now unused...

Reply:
So I'm stopping replications as the spec says to do so to create conflicting mutations. I think it makes sense at this stage to ensure we don't have timing issues etc. But I do agree we should have some concurrent updates but that maybe should be separate ticket.

Reply:
Definitely a separate ticket!

Reply:
Filed [CBG-4380](https://jira.issues.couchbase.com/browse/CBG-4380) for it!